### PR TITLE
Add CI test for pixel backend

### PR DESCRIPTION
**Pull request by d-uzlov**
Closes https://github.com/mlcommons/mobile_app_open/issues/457

### CODE REVIEW

**Comment by d-uzlov** [comment]
This PR consists of 2 parts:

1. Makefile refactoring (mostly android.mk)
2. Cloudbuild config refactoring

I have created a new cloudbuild job `android-build` to replace current `Build-and-test-Flutter-Android-App-PR` one.
With this new job firebase tests run in separate jobs.

**Review by anhappdev — CHANGES_REQUESTED** [review]

File: flutter/android/android-docker.mk [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		-v mlperf-mobile-flutter-cache-bazel-${flutter_docker_postfix}:/mnt/cache \
+		--env WITH_TFLITE=${WITH_TFLITE} \
+		--env WITH_QTI=${WITH_QTI} \
+		--env WITH_SAMSUNG=${WITH_SAMSUNG} \
+		--env WITH_PIXEL=${WITH_PIXEL} \
+		--env WITH_MEDIATEK=${WITH_MEDIATEK} \
+		--env BAZEL_ARGS_GLOBAL="${proxy_bazel_args} --output_user_root=/mnt/cache/bazel" \
+		--env OFFICIAL_BUILD=${OFFICIAL_BUILD} \
+		--env FLUTTER_FORCE_PUB_GET=1 \
+		${proxy_docker_args} \
```
[the remark this answers is not in the document]
Comment (d-uzlov) [reply flutter/android/android-docker.mk:48]: Done

File: flutter/flutter.mk [hunk 2]
```diff
@@ -32,9 +31,21 @@ flutter: flutter/prepare flutter/platform
...
 OFFICIAL_BUILD?=false
-flutter_official_build_flag=--dart-define=official-build=${OFFICIAL_BUILD}
+flutter_official_build_arg=--dart-define=official-build=${OFFICIAL_BUILD}
+.PHONY: flutter/check/official-build
+flutter/check/official-build:
+	@[ "$$OFFICIAL_BUILD" = "true" ] || [ "$$OFFICIAL_BUILD" = "false" ] \
+		|| (echo OFFICIAL_BUILD env must be explicitly set to \"true\" or \"false\"; exit 1)
+
+FLUTTER_BUILD_NUMBER?=0
+flutter_build_number_arg=--build-number ${FLUTTER_BUILD_NUMBER}
```
[the remark this answers is not in the document]
Comment (d-uzlov) [reply flutter/flutter.mk:44]: Done

**Review by anhappdev — APPROVED** [review]

**Review by anhappdev — CHANGES_REQUESTED** [review]
I got this error
```shell
➜ make docker/flutter/android/release
OFFICIAL_BUILD env must be explicitly set to "true" or "false"
make: *** [flutter/flutter.mk:40: flutter/check/official-build] Error 1
```

**Comment by d-uzlov** [comment]
@anhappdev
Like the error says, you should set the `OFFICIAL_BUILD` env to some value to fix it.
Either as a real env or as makefile env. For example, `make docker/flutter/android/release OFFICIAL_BUILD=false`.
In the makefile there is a default value for this env, but specifically for `release` scripts there is an additional check `flutter/check/official-build` to make sure that value is set explicitly and the default is not used.

By the way, similar error will happen with the `FLUTTER_BUILD_NUMBER` env.
Both are required for `docker/flutter/android/release`, `flutter/android/release`, and `flutter/ios/release` targets.

**Comment by anhappdev** [comment]
Ok. I got your point.

There is still a reference to the target `docker/flutter/android/apk` in the docs:
https://github.com/mlcommons/mobile_app_open/blob/e75502e3f04a3d08d19e3558c27043c3c7abdc29/flutter/documentation/build-and-run.md?plain=1#L53

**Comment by d-uzlov** [comment]
I'll fix the documentation.

**Review by anhappdev — DISMISSED** [review]

**Review by anhappdev — APPROVED** [review]

**Pull request merged**